File: maxro-backend/src/main/java/com/maxro/maxro_backend/service/AiChatService.java
package com.maxro.maxro_backend.service;

import com.maxro.maxro_backend.dto.ai.AiChatRequest;
import com.maxro.maxro_backend.dto.ai.AiChatResponse;
import com.maxro.maxro_backend.model.User;

public interface AiChatService {

    AiChatResponse reply(User user, AiChatRequest request);
}

File: maxro-frontend/src/app/core/services/ai-chat.service.ts
import { HttpClient, HttpHeaders } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable, throwError } from 'rxjs';
import { environment } from '../../../environments/environment';
import { AiChatRequest, AiChatResponse } from '../models/ai-chat.model';
import { AuthService } from './auth.service';

@Injectable({ providedIn: 'root' })
export class AiChatService {
  constructor(
    private readonly http: HttpClient,
    private readonly authService: AuthService,
  ) {}

  sendMessage(request: AiChatRequest): Observable<AiChatResponse> {
    const token = this.authService.getAccessToken();
    if (!token) {
      return throwError(() => new Error('Please sign in again to use chat.'));
    }

    const headers = new HttpHeaders({
      Authorization: `Bearer ${token}`,
      'Content-Type': 'application/json',
    });
    const apiBaseUrl = environment.apiBaseUrl || '';

    return this.http.post<AiChatResponse>(`${apiBaseUrl}/api/ai/chat`, request, { headers });
  }
}
